Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -380,9 +387,44 @@ def checkFileLength (lines : Array String) (existingLimit : Option ℕ) : Option
 
 end
 
+
+section unicodeLinter
+
+/- Hashable instance for use with the unicodeLinter -/
+local instance : Hashable Char where
+  hash c := c.val.toUInt64
+
+/-- TODO make complete and order nicely -/
+def unicodeWhitelist : Lean.HashSet Char := Lean.HashSet.ofList $ String.toList "
+Š≰⊈µ⋮ỳýì™⥥▼°𝓖𝔣и⅓ầ≣⇘č⚠чᴜᶹᵧ⋁´ᴄᴮᶻꜰßᴢᴏᴀⱽø𝐓ꜱᵋ½ᴰ⁼ɴÁꟴꞯăᶿᴶʟʜ𐞥ᵟ\u200bʙΒᵪᵝᵩᵘʳᵦʲᴺᴊᴛᴡᴼᴠ𝔇ɪ̀ᴿᵡᴱᴇоᴍʀᵂᴅᴷᴾɢʏᴵᵠᴘ−₌ĝᵨᴋᵞ⟍⁸н◥◹⁵◢
+◿ᶥ𝕋⟋⁹ńаИ⌟ł⊸₍ê′₎⁽ś⁾↻𝐒óŽ️№Łᴳ⁷𝔮̂▽꙳⟵в＼\u200c
+／æćà↠≱𝓦𝑹₉⁴⁶𝔽ʰ∖ř≟𝑳∋ᵗ—⊛É𝒵𝕌𝒅∤³⊃↟↘△↭∇𝓓ôᵏ¥𝓚𝔭𝓥𝕀𝓣𝔹┌--'
+⇉’◫ç⋈ïōš‼∥èℋő∯－ℐ𝕄ᵛ⊙𝓒ü⚬𝓑ℑ𝒥𝒯ℰ⍟𝓡ℜυ𝒫íò§Č∼𝓐ʷˡ𝔗𝒪├𝒱□✶ʸ⋊𝒄♩±𝒟†𝔻✝ᴹ𝒞⊚…𝒴ϖ₇↿–ä⊇𝐞ᴬ≼ᵍ∮₋𝔠↗⁄￢─“”𝒮𝕆áᵉ↾ᵁᴸ⩿⇔𝕝∐𝕊²⇒◃ⱼ
+⊨Ι⤳𝔼ℳₒ⨿𝓢Θⁱ𝓜○⧏⋯𝒳ₕ𝖣⬝ᵣ↓⇨ᘁ⟂⊻⊼⨯ₑ♯≺∠⨍₈∗⋖⊞ᴴₚ⨳∆⊂ᗮ∙₆ ⟹⋀⁺⌈⌉ᵀ∡𝕎𝓞ℙ⨂↥𝒢⅟ₖ│⊣«»𝟭ℓⁿ₅ℵℬ⨁𝓕ξ≈ᵇ≌⌊⌋ᵤ⟯⟮ℍ𝔸▷𝓘⊕𝓟≡ₓ◁ΛΔ𝕂
+√𝓤⁰ᵃ𝒰ϕ𝔖⋆Ψ↪öℱλ≪Σ⋂Π⟪é⟫⁆⁅ëΦ⨅ₜ⟦⟧ℒˣ‹›~ζ&θχ⨆⇑⧸⦃⦄∉₊⊢ₘᵈᵢₛˢ𝒜ₙᶜᵥ⋃ρ₄%∅ψ∪η⊔Ωᶠ∨∣ₐ∏⊓ℚ≅ᵖω𝟙ᵐ⊥Γτ⋙κ⊗¬∩▸ν⊤∫∂ᵒ∑ℂ\\
+∞δ∘≥𝓝⥤⊆↑ℤ≃ₗε∧σγ∃×¹79π6⁻8₃$5≠≫φ•⟶!4J↦‖₀↔𝕜ℕ3QZ^ℝYV;ι?μ≤K∈∀\"β←+W·<UX|@⟩⟨₁G₂*B→2H>D01OjNαz{}kEP/#'
+LTRAFqIMCw-][S`x,=vy()bg:.hdufcp\n
+ml_srantoie
+"
+
+/-- Lint a collection of input strings if one of them contains unwanted unicode. -/
+def unicodeLinter : TextbasedLinter := fun lines ↦ Id.run do
+  let mut errors : Array (StyleError × ℕ) := Array.mkEmpty 0
+  let mut lineNumber := 1
+  for line in lines do
+    let badChars := line.toList.filter (fun c => !unicodeWhitelist.contains c)
+    if badChars.length > 0 then
+      let newErrors : Array (StyleError × ℕ) :=
+        badChars.toArray.map fun c ↦ (StyleError.newUnicode c, lineNumber)
```

Comment:
Here's a small Lean trick, just so you know.
You can also use \centerdot to write this more compactly as `badChars.toArray.map (StyleError.newUnicode ·, lineNumber)`; same for the definition of `badChars` three lines higher.